core/dbt/include/global_project/macros/materializations/snapshots/snapshot.sql

```diff
@@ -26,7 +26,7 @@
 
   {% if not target_relation_exists %}
 
-      {% set build_sql = build_snapshot_table(strategy, model['compiled_sql']) %}
+      {% set build_sql = build_snapshot_table(strategy, model['compiled_code']) %}
```

Comment:
@jtcohen6 I am not sure whether this is used in other places but sort of a breaking change we got here

Reply:
we can fix adapters we maintain if there's a issue but hard to do for third party ones

Reply:
@ChenyuLInx Good find. This is pretty uncommon behavior, but it's true that we have [the same code](https://github.com/dbt-labs/dbt-spark/blob/244742ac0ee35db9774d9830a96d5796bf738448/dbt/include/spark/macros/materializations/snapshot.sql#L120) in `dbt-spark` right now.

Could we preserve `raw_sql` + `compiled_sql` as node attributes / aliases for their `_code` counterparts, without including them in `manifest.json` output?

Otherwise, we'll just need to call this out as a breaking change. It should be one with limited blast radius.

Reply:
Yea, we can do that since we just need to do some custom logic in the provider to do it when generating the context. And maybe put some depreciation warnings to remove it in 2.0?

Reply:
> And maybe put some depreciation warnings to remove it in 2.0?

Good call. Happy to make that piece a follow-on ticket, rather than expand the scope of this PR. Could be a nice exercise for someone on the team who hasn't yet gotten to touch our [`deprecations` module](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/deprecations.py)

Reply:
This has been updated and #5558 is created  to track it